core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -885,55 +897,47 @@ private boolean shouldWrite(StoreStatus storeStatus, Predicate<? super StoreConf
-            },
-            this::releaseReadLock
-      ).toCompletionStage(null);
+      //noinspection unchecked
+      return batchOperation(txInvocationContext, null, (stores, segmentCount, removeFlowable,
+            putFlowable) -> stores.prepareWithModifications(txInvocationContext.getTransaction(), segmentCount, removeFlowable, putFlowable))
+            .thenApply(CompletableFutures.toNullFunction());
    }
 
    @Override
-   public CompletionStage<Void> commitAllTxStores(Transaction transaction, Predicate<? super StoreConfiguration> predicate) {
+   public CompletionStage<Void> commitAllTxStores(TxInvocationContext<AbstractCacheTransaction> txInvocationContext,
+         Predicate<? super StoreConfiguration> predicate) {
       return Completable.using(
             this::acquireReadLock,
             ignore -> {
                checkStoreAvailability();
                if (trace) {
-                  log.tracef("Committing transaction %s to stores", transaction);
+                  log.tracef("Committing transaction %s to stores", txInvocationContext);
                }
                return Flowable.fromIterable(stores)
                      .filter(storeStatus -> shouldPerformTransactionOperation(storeStatus, predicate))
                      // Let the commit work in parallel across the stores
-                     .flatMapCompletable(storeStatus -> Completable.fromCompletionStage(storeStatus.store.commit(transaction)));
+                     .flatMapCompletable(storeStatus -> Completable.fromCompletionStage(storeStatus.store.commit(txInvocationContext.getTransaction())));
             },
             this::releaseReadLock
       ).toCompletionStage(null);
    }
 
    @Override
-   public CompletionStage<Void> rollbackAllTxStores(Transaction transaction, Predicate<? super StoreConfiguration> predicate) {
+   public CompletionStage<Void> rollbackAllTxStores(TxInvocationContext<AbstractCacheTransaction> txInvocationContext,
+         Predicate<? super StoreConfiguration> predicate) {
       return Completable.using(
             this::acquireReadLock,
             ignore -> {
                checkStoreAvailability();
                if (trace) {
-                  log.tracef("Rolling back transaction %s for stores", transaction);
+                  log.tracef("Rolling back transaction %s for stores", txInvocationContext);
                }
                return Flowable.fromIterable(stores)
                      .filter(storeStatus -> shouldPerformTransactionOperation(storeStatus, predicate))
                      // Let the rollback work in parallel across the stores
-                     .flatMapCompletable(storeStatus -> Completable.fromCompletionStage(storeStatus.store.commit(transaction)));
+                     .flatMapCompletable(storeStatus -> Completable.fromCompletionStage(storeStatus.store.commit(txInvocationContext.getTransaction())));
```

Comment:
Shouldn't this be `storeStatus.store.rollback`?

It's worrying that this wasn't detected by our testsuite :scream:

Reply:
:( My guess it the JDBC store just doesn't have great coverage around this case. I can fix this and I will log a JIRA to create a test case for JDBC store.

Reply:
Fixed the typo